From 9e87280e0db89cece5ad2ce743221b1789dfd42c Mon Sep 17 00:00:00 2001
From: David Wu <david.wu@rock-chips.com>
Date: Tue, 23 Feb 2016 20:14:45 +0800
Subject: [PATCH] keys: rockchip: use late_initcall for driver init

Use late_initcall to ensure saradc driver init bofore
keys driver init.

Change-Id: I34a12cb4d3e28c749d9a8bbf35329185edf7c0c6
Signed-off-by: David Wu <david.wu@rock-chips.com>
---
 drivers/input/keyboard/rk_keys.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/drivers/input/keyboard/rk_keys.c b/drivers/input/keyboard/rk_keys.c
index 1071b54a454c..8e502eea1642 100755
--- a/drivers/input/keyboard/rk_keys.c
+++ b/drivers/input/keyboard/rk_keys.c
@@ -534,4 +534,15 @@ static struct platform_driver keys_device_driver = {
 	}
 };
 
-module_platform_driver(keys_device_driver);
+static int __init rk_keys_driver_init(void)
+{
+	return platform_driver_register(&keys_device_driver);
+}
+
+static void __exit rk_keys_driver_exit(void)
+{
+	platform_driver_unregister(&keys_device_driver);
+}
+
+late_initcall_sync(rk_keys_driver_init);
+module_exit(rk_keys_driver_exit);
-- 
2.35.3

